module1-introduction-to-backend/r4-server-side-validation/README.md

```diff
@@ -0,0 +1,136 @@
+# Server-side Validation
+Your API can handle different types of requests now. But does it validate the request data before processing it? The objectives of this lesson are:
+1. Understanding the need for server-side validation
+2. Getting familiar with implementing validation
+3. Understanding the characteristics of a good error response
+
+## What is Validation?
+Validation can mean a lot of things, but in API land it generally means figuring out if the data being sent to the API is any good or not. Validation can happen both on client-side before sending the request or on server-side when receiving the request. Client-side validation is generally used to provide quick feedback to a user. For example, when you're submitting a form and you see a field highlighted in red saying "Required field". Basically the client validated that the field cannot be empty. However, an API must not entirely rely on client-side validation. We have talked about how the frontend need not know the underlying implementation of an API. Similarly, an API does not know what actually happened on the frontend when it receives a request from a client. That is why more often than not, the first step in processing a request is to validate the data that came with it.
```

Comment:
you should mention that one shoudl not rely purely on frontend validation, for example if you only validate on the frontend, someone can simply use postman to bypass such validation. almost all live apps will have validation in both places (despite the redundancy -- one for good UI, the other to make sure that there's no way to circumvent validation)

i see you said they should not rely "entirely on client-side validation" but IMO the main reason to do serverside is actually more for security reasons (in addition to the fact that, as you mentioned, not all things are known on the client)